test/test_autograd.py

```diff
@@ -4129,6 +4130,863 @@ def fn(*inputs):
         setattr(TestAutogradDeviceType, test_name, do_test)
 
 
+class TestAutogradFunctional(TestCase):
+    def _assert_same_struct(self, res, base):
+        # base and res should be Tensors or tuple of Tensors containing the same thing
+        if torch.is_tensor(base):
+            self.assertTrue(torch.is_tensor(res))
+            self.assertTrue(base.size() == res.size())
```

Comment:
Does `self.assertEqual(base.size(), res.size())` work? The error message for assertTrue is not the best (it doesn't help when debugging problems in the CI because it just prints "Assertion error: expected True but got False")

Reply:
I can try it. I'll let you know if it works.

Reply:
Seems to work !